server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -553,6 +553,22 @@ private void postCreateRecurringSnapshotForPolicy(long userId, long volumeId, lo
         }
     }
 
+    public void markRemovedSnapshot(long snapshotId) {
```

Comment:
_markRemovedSnapshot_ indicates the snapshot is already removed. I think _removeSnapshot_ (or _markSnapshotAsRemoved_) is ok.